esp-hal/src/dma/buffers.rs

```diff
@@ -1562,3 +1564,364 @@ impl DerefMut for DmaLoopBuf {
         self.buffer
     }
 }
+
+/// A Preparation that masks itself as a DMA buffer.
+///
+/// Fow low level use, where none of the pre-made buffers really fit.
+///
+/// This type likely never should be visible outside of esp-hal.
+pub(crate) struct NoBuffer(Preparation);
+impl NoBuffer {
+    fn prep(&self) -> Preparation {
+        Preparation {
+            start: self.0.start,
+            direction: self.0.direction,
+            #[cfg(psram_dma)]
+            accesses_psram: self.0.accesses_psram,
+            burst_transfer: self.0.burst_transfer,
+            check_owner: self.0.check_owner,
+            auto_write_back: self.0.auto_write_back,
+        }
+    }
+}
```

Comment:
Interesting, I suppose this would be shorter 😄
```Rust
pub(crate) struct NoBuffer(Option<Preparation>);
impl NoBuffer {
    fn prep(&mut self) -> Preparation {
        unwrap!(self.0.take());
    }
}
```

The unwrap is not ideal I guess.